src/Message/AbstractRequest.php

```diff
@@ -1,54 +1,55 @@
 <?php
+declare(strict_types=1);
 
 namespace MyOnlineStore\Omnipay\KlarnaCheckout\Message;
 
-use Guzzle\Common\Event;
-use Guzzle\Common\Exception\InvalidArgumentException;
-use Guzzle\Common\Exception\RuntimeException;
-use Guzzle\Http\ClientInterface;
-use Guzzle\Http\Exception\RequestException;
-use Guzzle\Http\Message\RequestInterface;
-use Guzzle\Http\Message\Response;
-use MyOnlineStore\Omnipay\KlarnaCheckout\AuthenticationRequestOptionProvider;
+use Money\Currency;
+use Money\Money;
+use MyOnlineStore\Omnipay\KlarnaCheckout\AuthenticationRequestHeaderProvider;
 use MyOnlineStore\Omnipay\KlarnaCheckout\CurrencyAwareTrait;
 use MyOnlineStore\Omnipay\KlarnaCheckout\ItemBag;
-use Symfony\Component\HttpFoundation\Request as HttpRequest;
+use Omnipay\Common\Message\AbstractRequest as BaseAbstractRequest;
+use Psr\Http\Message\ResponseInterface;
 
 /**
  * @method ItemBag|null getItems()
  */
-abstract class AbstractRequest extends \Omnipay\Common\Message\AbstractRequest
+abstract class AbstractRequest extends BaseAbstractRequest
 {
     use CurrencyAwareTrait;
 
     /**
-     * @inheritdoc
+     * @return Money|null
      */
-    public function __construct(ClientInterface $httpClient, HttpRequest $httpRequest)
+    public function getAmount()
     {
-        parent::__construct($httpClient, $httpRequest);
-
-        // don't throw exceptions for 4xx errors
-        $this->httpClient->getEventDispatcher()->addListener(
-            'request.error',
-            function (Event $event) {
-                if ($event['response']->isClientError()) {
-                    $event->stopPropagation();
-                }
-            }
-        );
+        if (null === $amount = $this->getParameter('amount')) {
+            return null;
+        }
+
+        if ($amount instanceof Money) {
+            return $amount;
+        }
+
+        try {
+            $currency = new Currency($this->getCurrency());
+        } catch (\InvalidArgumentException $exception) {
+            $currency = new Currency('USD');
+        }
+
+        return new Money($amount, $currency);
```

Comment:
This fails if amount is passed in decimal string format (which is currently the case). You'll probably need to use the DecimalMoneyParser here as well.

Same for `getTaxAmount`